pages/design_documentation/cleaner.md

```diff
@@ -5,12 +5,148 @@ toc: false
 permalink: cleaner.html
 ---
 
-<small>[[TODO](/authoring.html)] Add more thorough description of the cleaner and policies</small>
+Cleaner is a necessary feature of any caching solution. In the end, if there is
+[**dirty data**](/cache_line.html#valid-and-dirty-bits), it must always be 
+[**cleaned**](/cleaning.html) by flushing the dirty data from the cache to a 
+backing storage.
+
+In the Open CAS Framework, depending on a selected cache mode and a particular 
+use case, data is cleand in different situations and by different means. For 
+example, the dirty data may be cleaned in the backround by the cleaner task or 
+the cleaning may be triggered by an I/O engine. Moreover, dirty data must be 
+cleaned in order to perform eviction, when cache is 100% full and the data is 
+100% dirty. On top of that, user can also request a manual data flush by calling 
+appropriate cache management function.
+
+## Background cleaning
+The Open CAS Framework implements different strategies of determining when and what 
+data is subject to being cleaned by the background cleaner. These strategies are 
+different [**cleaning policies**](/cleaner.html#cleaning-policies). Cleaning policy 
+is set by the user upon creating a cache.
+
+During cache start up, a user selected cleaning policy is applied. If not explicitly 
+specified, a cleaning policy of [**ALRU**](/cleaner.html#cleaning-policies) is selected by default.
+
+The cleaning is performed by periodically invoking *ocf_cleaner_run()* by the 
+OCF user. User is expected to call it as a response to the *kick()* callback and 
```

Comment:
In the previous paragraph you used "user" in context of a person using CAS and here you mean an adapter. To avoid confusion I suggest to use the word "adapter" explicitly.